test/functorch/test_aotdispatch.py

```diff
@@ -2923,23 +2865,12 @@ def _test_aot_autograd_helper(self, device, dtype, op, dynamic=False):
     for sample_input in sample_inputs_itr:
         t_args = [sample_input.input] + list(sample_input.args)
         t_kwargs = sample_input.kwargs
-        flat_args, args_spec = pytree.tree_flatten((t_args, t_kwargs))
-        sentinel_val = -42
-        is_tensor_spec = [sentinel_val if isinstance(arg, torch.Tensor) else arg for arg in flat_args]
-        args = [arg for arg in flat_args if isinstance(arg, torch.Tensor)]
-
-        def f(args):
-            cur_flat_args = list(is_tensor_spec)
-            args = iter(args)
-            for idx, v in enumerate(cur_flat_args):
-                if v == sentinel_val:
-                    cur_flat_args[idx] = next(args)
-            c_args, c_kwargs = pytree.tree_unflatten(cur_flat_args, args_spec)
-            return op.op(*c_args, **c_kwargs)
-
-        compiled_f = compiled_function(f, nop, nop, dynamic=dynamic, partition_fn=min_cut_rematerialization_partition)
         try:
-            _test_aot_autograd_forwards_backwards_helper(self, f, compiled_f, args)
+            aot_autograd_check(
+                op.op, t_args, t_kwargs, dynamic,
+                self.assertRaisesRegex, self.assertEqual)
+        except DynamicOutputShapeException:
+            self.skipTest("Dynamic output shape operation in trace")
```

Comment:
hmm any reason we had to add the skip here? IIRC, we'd previously just xfail'd the list of opinfos that we knew would cause dynamic shape exceptions during tracing.

Reply:
This skip is pre-existing and used to be in `_test_aot_autograd_forwards_backwards_helper`. I moved it up to this function.

Reply:
whoops I missed that. Thanks!